### PR TITLE
Automate Python docs.ms onboarding

### PR DESCRIPTION
**Pull request by danieljurek**
Automates onboarding of packages for docs.ms CI. Includes validation to ensure that new and updated packages install properly. Does not re-validate packages that have not changed. 

docindex successful run (pushing against a test branch, not main): https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1030266&view=logs&j=293d82b9-e70f-547c-66fb-8f46778b3a20

docs CI running against test branch: https://apidrop.visualstudio.com/Content%20CI/_build/results?buildId=244126&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=e69637ba-916f-5e3a-caa0-a58126252db6

### CODE REVIEW

**Comment by danieljurek** [comment]
@scbedd PTAL

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -161,68 +161,200 @@ function Get-python-GithubIoDocIndex()
...
-{
-  $pkgJsonLoc = (Join-Path -Path $ciRepo -ChildPath $locationInDocRepo)
+function ValidatePackage($packageName, $packageVersion, $workingDirectory) {
+  $packageExpression = "$packageName$packageVersion"
+  Write-Host "Validating $packageExpression"
+
+  $installTargetFolder = Join-Path $workingDirectory $packageName
+  New-Item -ItemType Directory -Force -Path $installTargetFolder | Out-Null
+
+  # TODO: Additional checks for package validation in docs build.
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:171]: Is this a real TODO with a specific set of things you want to add? If so please file an issue to track it, if not probably remove the TODO word.

Comment (danieljurek) [reply eng/scripts/Language-Settings.ps1:171]: I think we should replicate as much of the docs CI process as we possibly can for each package that we want to test. It'll give us the greatest confidence that the docs CI process will build docs artifacts.

**Review by weshaggard — APPROVED** [review]
Looks reasonable

**Pull request merged**